source/ShipJumpNavigation.cpp

```diff
@@ -87,6 +88,28 @@ double ShipJumpNavigation::JumpFuel(const System *destination) const
 
 
 
+// Get the amount of fuel that would be expended to jump to the "nearest" (in fuel) system.
+double ShipJumpNavigation::JumpFuelNearest() const
+{
+	// A currently-carried ship or ship without drives requires no fuel to jump,
+	// because it cannot jump.
+	if(!currentSystem || !HasAnyDrive())
+		return 0.;
+
+	double jumpDriveCost = JumpDriveFuel();
+	double hyperdriveCost = HyperdriveFuel();
+
+	// Check whether the ship can jump, that jumping is more expensive than hyperdriving,
+	// and whether the system has links.
+	if((!jumpDriveCost || jumpDriveCost >= hyperdriveCost) && !currentSystem->Links().empty())
+		return hyperdriveCost;
```

Comment:
I think you also need to check that `hyperdriveCost` isn't zero? Else, if you have no hyperdrive but you do have a jump drive, this will give zero, which doesn't seem right.

Reply:
Yeah I forgot to account for that,thought that `jumpDriveCost >= hyperdriveCost` would cover it and forgetting that it was inside an OR expression. Fixed.